subprojects/integ-test/src/integTest/groovy/org/gradle/integtests/SyncTaskIntegrationTest.groovy

```diff
@@ -394,36 +393,70 @@ class SyncTaskIntegrationTest extends AbstractIntegrationSpec {
         ins.close()
     }
 
-    @ToBeImplemented
     @Issue("https://github.com/gradle/gradle/issues/9586")
-    @Requires(TestPrecondition.CASE_INSENSITIVE_FS)
-    def "change in case of input file still syncs properly"() {
+    def "change in case of input files and folders will still syncs properly"() {
         given:
-        buildFile << """
-            task writeFile {
-                def outputFile = new File(buildDir, project.hasProperty("capitalize") ? "OUTPUT" : "output")
-                outputs.file outputFile
-                doLast {
-                    outputFile.text = "hello"
-                }
-            }
-            task sync(type: Sync) {
-                from writeFile
-                into new File(buildDir, "sync")
+        buildFile << '''
```

Comment:
using the same formatting rules as the rest of the file: `'` for Strings, sync task called `syncIt`, folder creation with `create`.
Left `buildFile << ` instead of `buildScript`, since @lptr explicitly stated he likes this format more.